### PR TITLE
Add support for Foxy

### PR DESCRIPTION
**Pull request by esteve**
This PR adds support for Foxy, since it'll be supported for longer than Galactic (November 2022 vs May 2023), Foxy is the latest LTS until Humble Hawksbill comes out next year. Additionally, Autoware.Auto is still on Foxy.

### CODE REVIEW

File: serial_driver/src/serial_bridge_node.cpp [hunk 1]
```diff
@@ -126,21 +126,36 @@ void SerialBridgeNode::get_params()
   auto sb = StopBits::ONE;
 
   try {
+// NOTE(esteve): remove this when Foxy is no longer supported
+#if RCLCPP_VERSION_MAJOR > 2
     m_device_name = declare_parameter<std::string>("device_name");
+#else
+    m_device_name = declare_parameter<std::string>("device_name", "");
+#endif
```
Comment (JWhitleyWork) [inline serial_driver/src/serial_bridge_node.cpp:134]: Couldn't you just use the defaulted signature for all ROS versions and avoid the macros?

Comment (esteve) [reply serial_driver/src/serial_bridge_node.cpp:134]: That's fine too, I just wanted to keep the code as close as possible to the current state for Galactic.

Comment (esteve) [reply serial_driver/src/serial_bridge_node.cpp]: Done in 5c43956

Comment (JWhitleyWork) [reply serial_driver/src/serial_bridge_node.cpp]: Thanks!

**Pull request merged**